src/component.py

```diff
@@ -130,25 +128,31 @@ def get_stats_for_date(self, session, date, eshop_id):
                 'NÃ¡klady zÂ obratu': 'pno',
             }
 
-        column_names = [th.text for th in response.html.find('thead', first=True).find('tr')[1].find('th')]
-        table_body = response.html.find('tbody', first=True)
+        try:
+            column_names = [th.text for th in response.html.find('thead', first=True).find('tr')[1].find('th')]
+            table_body = response.html.find('tbody', first=True)
+
+            if table_body:
 
-        if table_body:
+                values = [value.text.replace('Â\xa0KÄ\x8d', '').replace('Â â\x82¬', '').replace('%', '')
+                          .replace('Â', '').replace(' ', '').replace('&nbsp', '').replace(' ', '')
+                          for value in table_body.find('tr')[0].find('td')]
 
-            values = [value.text.replace('Â\xa0KÄ\x8d', '').replace('Â â\x82¬', '').replace('%', '')
-                      .replace('Â', '').replace(' ', '').replace('&nbsp', '').replace(' ', '')
-                      for value in table_body.find('tr')[0].find('td')]
+                row = {'eshop_id': eshop_id, 'date': date["start_date"]}
 
-            row = {'eshop_id': eshop_id, 'date': date["start_date"]}
+                if values[0] == 'Celkem':
+                    logging.warning("No data available for the selected period")
+                else:
+                    for column_name, value in zip(column_names, values):
+                        if key := columns_mapping.get(column_name):
+                            row[key] = value
 
-            if values[0] == 'Celkem':
-                logging.warning("No data available for the selected period")
-            else:
-                for column_name, value in zip(column_names, values):
-                    if key := columns_mapping.get(column_name):
-                        row[key] = value
+                return row
 
-            return row
+        except AttributeError as e:
+            self.login()
+            logging.warning("Table not found, logging in again")
+            raise e
```

Comment:
Tady by mělo být něco jinýho, než AttributeError, ne? TableNotFoundException, nebo něco takovýho.

Reply:
jedná se o toto: `AttributeError: 'NoneType' object has no attribute 'find` a protože tam je ten find použitý několikrát jako např.: `response.html.find('thead', first=True).find('tr')[1].find('th')]` přijde mi toto lepší než řešit na každé úrovni zda element existuje

Reply:
Já to myslel tak, že by ji to nererasiovalo v tom `raise e`, ale spíš něco, co bude vyjadřovat to, že to nenašlo tu tabulku.